lsp-io.el

```diff
@@ -99,23 +99,27 @@
       'notification
       (signal 'lsp-unknown-message-type (list json-data)))))
 
+(defconst lsp--default-notification-handlers
+  #s(hash-table
+     test eq
```

Comment:
Should the test here be `equal`? `eq` only checks for reference equality, not object/structural equality:

```emacs-lisp
(setq h (make-hash-table :test 'eq))
(puthash "foo" 'bar h)
(gethash "foo" h)
```

evaluates to `nil`.